### PR TITLE
build files: add dependency on libgrpc for projects linking libgrpc++

### PR DESCRIPTION
**Pull request by emmenlau**
On MINGW its required that when linking to `libgrpc++`, to also link `libgrpc`. I would assume that this is required for most platforms, but maybe this takes place implicitly? In any case, I have modified build.yaml so that wherever `grpc++` is listed in `dep:`, also `grpc` is listed in `dep:`.

### CODE REVIEW

**Comment by jtattermusch** [comment]
please resolve conflicts

File: build.yaml [hunk 1]
```diff
@@ -3187,8 +3194,8 @@ targets:
   deps:
```
Comment (jtattermusch) [inline build.yaml]: Btw, the order of dependencies in the original code 
https://github.com/grpc/grpc/blob/6b6954050cf0379dbbee90dd18313a3779e0dd52/build.yaml#L3204 (it should be mostly the reverse, as for all the other targets).

**Comment by emmenlau** [comment]
I have fixed the conflicts, and changed to order of dependencies. To fix the conflicts, I hopefully
did it in a clean way: while merging I regenerated the auto-generated build files and overwrote
the conflicting files, then commited the updated merge. Is that ok/good practice?

**Comment by jtattermusch** [comment]
@emmenlau  yes, rewriting the generated files is a good practice.

File: build.yaml [hunk 2]
```diff
@@ -1230,8 +1235,8 @@ libs:
...
   - grpc++_test_util
   - grpc_test_util
-  - grpc++
   - grpc
+  - grpc++
```
Comment (jtattermusch) [inline build.yaml]: The dependencies in build.yaml need to come in reverse order. (grpc++  comes before grpc)
(many occurences).

**Comment by jtattermusch** [comment]
Is this still needed?  (also, there are merge conflicts that would need to be resolved).
Closing for now, feel free to reopen.

**Comment by emmenlau** [comment]
Closed in favor of new PR https://github.com/grpc/grpc/pull/10075

**Pull request closed without merging**